### PR TITLE
[SPARK-30336][SQL][SS] Move Kafka consumer-related classes to its own package

### PR DESCRIPTION
**Pull request by HeartSaVioR**
### What changes were proposed in this pull request?

There're too many classes placed in a single package "org.apache.spark.sql.kafka010" which classes can be grouped by purpose.

As a part of change in SPARK-21869 (#26845), we moved out producer related classes to "org.apache.spark.sql.kafka010.producer" and only expose necessary classes/methods to the outside of package. This patch applies the same to consumer related classes.

### Why are the changes needed?

Described above.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Existing UTs.

### CODE REVIEW

File: external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceRDD.scala [hunk 1]
```diff
@@ -19,18 +19,15 @@ package org.apache.spark.sql.kafka010
 
 import java.{util => ju}
 
-import scala.collection.mutable.ArrayBuffer
```
Comment (HeartSaVioR) [inline external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceRDD.scala]: I've also cleaned up unused imports and remove unnecessary line (two empty lines between the last import statement and class definition)

File: external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/consumer/InternalKafkaConsumerPool.scala [hunk 2]
```diff
@@ -218,4 +218,3 @@ private[kafka010] object InternalKafkaConsumerPool {
     }
   }
 }
-
```
Comment (HeartSaVioR) [inline external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/consumer/InternalKafkaConsumerPool.scala]: There were two empty lines which can be just one empty line.

**Review by srowen — COMMENTED** [review]
Looks OK as a cleanup, pending tests

**Comment by HeartSaVioR** [comment]
retest this, please

**Comment by HeartSaVioR** [comment]
cc. @srowen Gently ping.

**Review by srowen — APPROVED** [review]
If tests pass and it's just a refactoring, OK. 
I guess you also made everything private to .consumer that you can?

**Comment by HeartSaVioR** [comment]
> I guess you also made everything private to .consumer that you can?

Yes, the KafkaDataConsumer itself, and acquire and release are open to other package (kafka010). Others are private to consumer.

**Comment by srowen** [comment]
Merged to master

**Comment by HeartSaVioR** [comment]
Thamks for reviewing and merging! Happy new year!

**Pull request closed without merging**